static/src/javascripts/projects/commercial/modules/dfp/create-slots.ts

```diff
@@ -194,61 +193,103 @@ const createAdSlotElements = (
 		adSlot.setAttribute(attr, attrs[attr]);
 	});
 
-	adSlots.push(adSlot);
+	return adSlot;
+};
 
-	return adSlots;
+/**
+ * Split class names and prefix all with ad-slot--${className}
+ */
+const createClasses = (slotName: string, classes?: string): string[] =>
+	[...(classes?.split(' ') ?? []), slotName].map(
+		(className: string) => `ad-slot--${className}`,
+	);
+
+/**
+ * Given default size mappings and additional size mappings from
+ * the createAdSlot options parameter.
+ *
+ * 1. Check that the options size mappings use known device names
+ * 2. If so merge the size mappings
+ *
+ */
+const mergeSizeMappings = (
```

Comment:
This function won't deduplicate size mappings where merging? Does a name like merge imply that it might?

Reply:
Maybe “join” is the more explicit word?

Reply:
Oh thats interesting, to me 'merge' doesn't imply dedupe but I can see that it could be misconstrued.
I've updated to 'concat' to reflect the actual operation - I hope that is clearer?

Reply:
I think concat makes the most sense, since at the end of the day we're concatenating the sizes at each breakpoint.